lib/tty/file.rb

```diff
@@ -309,10 +319,11 @@ def copy_file(source_path, *args, context: nil, force: false, skip: false,
         content = block[content] if block
         content
       end
-      return unless preserve
+      return file_dest unless preserve
 
       copy_metadata(source_path, dest_path, verbose: verbose, noop: noop,
                     color: color)
+      file_dest
```

Comment:
We need a test case that confirms this behaviour so that it stays true in future releases.